python/ray/dag/compiled_dag_node.py

```diff
@@ -112,7 +115,27 @@ def do_exec_tasks(
             if done:
                 break
             for operation in schedule:
+                start_t = time.time()
                 done = tasks[operation.local_idx].exec_operation(self, operation.type)
+                end_t = time.time()
+
+                if RAY_ADAG_ENABLE_PROFILING:
```

Comment:
nit: but can you move this to the top of the function and have a local variable instead?

```

@DeveloperAPI
def do_exec_tasks(
    self,
    tasks: List["ExecutableTask"],
    schedule: List[_DAGNodeOperation],
) -> None:
    """A generic actor method to begin executing the operations belonging to an
    actor. This runs an infinite loop to execute each _DAGNodeOperation in the
    order specified by the schedule. It exits only if the actor dies or an
    exception is thrown.
    Args:
        tasks: the executable tasks corresponding to the actor methods.
        schedule: A list of _DAGNodeOperation that should be executed in order.
    """
    enable_profiling = RAY_ADAG_ENABLE_PROFILING

```

the reason is it is usually more expensive to access global variable than local variable. since it is perf sensitive path, Id' like to make sure we don't introduce any additional overhead.

Reply:
Oh I see. Didn't expect that this is so performance sensitive. Great to learn and I'll change this.